qpc/tests/report/test_report_aggregate.py

```diff
@@ -0,0 +1,122 @@
+"""Test the CLI module."""
+
+import logging
+from argparse import ArgumentParser, Namespace
+from io import StringIO
+from unittest.mock import patch
+
+import pytest
+import requests_mock
+
+from qpc import messages
+from qpc.report import AGGREGATE_PATH_SUFFIX, REPORT_URI
+from qpc.report.aggregate import ReportAggregateCommand
+from qpc.scan import SCAN_JOB_URI
+from qpc.tests.utilities import redirect_stdout
+from qpc.utils import get_server_location
+
+pytestmark = [pytest.mark.usefixtures("server_config")]
+
+
+def get_scan_job_uri(scan_job_id: int) -> str:
+    """Construct a scan job URI for testing."""
+    return f"{get_server_location()}{SCAN_JOB_URI}{scan_job_id}"
+
+
+def get_aggregate_report_uri(report_id: int) -> str:
+    """Construct an aggregate report URI for testing."""
+    return f"{get_server_location()}{REPORT_URI}{report_id}{AGGREGATE_PATH_SUFFIX}"
+
+
+def get_command():
+    """Set up new command for each test."""
+    # We can't use a fixture for this, even though that seems reasonable,
+    # because our command instances modify req_path on the fly. This is
+    # definitely a bad code smell, but we are choosing to ignore it because
+    # this design issue affects many of our commands the same way.
+    argument_parser = ArgumentParser()
+    subparser = argument_parser.add_subparsers(dest="subcommand")
+    return ReportAggregateCommand(subparser)
+
+
+@patch("qpc.report.aggregate.pretty_format")
+def test_aggregate_report_success_via_report(mock_pretty_format, faker):
+    """Test aggregate report with report id."""
+    report_out = StringIO()
+
+    report_id = faker.pyint()
+    report_uri = get_aggregate_report_uri(report_id)
+    report_json_data = {faker.slug(): faker.slug()}
+    with requests_mock.Mocker() as mocker:
+        mocker.get(report_uri, status_code=200, json=report_json_data)
+        args = Namespace(scan_job_id=None, report_id=report_id)
+        with redirect_stdout(report_out):
+            get_command().main(args)
+        mock_pretty_format.assert_called_once_with(report_json_data)
+
+
+@patch("qpc.report.aggregate.pretty_format")
+def test_aggregate_report_success_via_scan_job(mock_pretty_format, faker):
+    """Test aggregate report with scan job id."""
+    report_out = StringIO()
+
+    report_id = faker.pyint()
+    scan_job_id = faker.pyint()
+    scan_job_uri = get_scan_job_uri(scan_job_id)
+    scan_job_json_data = {"id": scan_job_id, "report_id": report_id}
+    aggregate_report_uri = get_aggregate_report_uri(report_id)
+    aggregate_report_json_data = {faker.slug(): faker.slug()}
+    with requests_mock.Mocker() as mocker:
+        mocker.get(scan_job_uri, status_code=200, json=scan_job_json_data)
+        mocker.get(
+            aggregate_report_uri, status_code=200, json=aggregate_report_json_data
+        )
+        args = Namespace(scan_job_id=scan_job_id, report_id=None)
+        with redirect_stdout(report_out):
```

Comment:
I didn't follow why write stdout to `report_out`, if we aren't doing anything with it. Or are we and I missed it?

Same for other similar uses of `redirect_stdout`.

TBH I nitpick a bit with this utility and would love to 🪓 it. I'm slowly replacing it with `capsys` fixture, which also captures stderr.

Reply:
Maybe for unitttest it made sense to have it just to shush qpc while testing. Nowadays with pytest this is unnecessary.

Reply:
I started by wholesale duplicating an existing test module for another command, and I chopped away a bunch of the fluff that I didn't need. If I can chop this away too, I'm happy to do it! I'll take a second look at this part.

Reply:
Updated and pushed. Much better now, I think. Thanks for suggesting this one!